xenium-flash/src/xenium-flash.cpp

```diff
@@ -171,6 +171,7 @@ int main(int argc, char** argv)
     {
         //write byte to flash
         flash.Write(i, flash_buffer[i]);
+        while(flash.Read(0) != flash.Read(0));
```

Comment:
This is an infinite while loop.  

please provide X amount of retries with error condition if it fails..